modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheAdapter.java

```diff
@@ -3225,6 +3225,18 @@ public CacheMetricsImpl metrics0() {
         return txs.txStartEx(ctx, concurrency, isolation, timeout, txSize).proxy();
     }
 
+    /** {@inheritDoc} */
+    @Override public GridNearTxLocal txStartEx(
```

Comment:
Introducing these methods doesn't make sense to me.
You can unwrap local tx using the following code: 

`GridNearTxLocal tx0 = ((TransactionProxyImpl)tx).tx();`

Reply:
In my opinion, this code does not look good enough. Moreover, it requires creating a new `TransactionProxyImpl` instance which is redundant I think.